validator/src/scheduler/mod.rs

```diff
@@ -52,6 +52,7 @@ pub struct ExecutionResults {
     pub beginning_state_hash: Option<String>,
     pub ending_state_hash: Option<String>,
     pub batch_results: Vec<(String, Vec<TxnExecutionResult>)>,
+    pub unscheduled_batch_ids: Vec<String>,
```

Comment:
Instead of adding a new field, I think you can include batches that were not executed in batch_results by making it something like `Vec<(String, Option<Vec<TxnExecutionResult>>)>`. You probably want to introduce a new type at that point though, so you could either do `Vec<BatchExecutionResult>` where `struct BatchExecutionResult { batch_id: String, result: Option<Vec<TxnExecutionResult>> }` or `Vec<(String, Option<BatchExecutionResult>)>` where `type BatchExecutionResult = Vec<TxnExecutionResult>`